pkg/api/norman/customization/cluster/validator.go

```diff
@@ -601,7 +601,11 @@ func validateEKSKubernetesVersion(spec *v32.ClusterSpec, prevCluster *v3.Cluster
 }
 
 // validateCredentialAuth validates that a user has access to the credential they are setting.
-func validateCredentialAuth(request *types.APIContext, credential string) error {
+func validateCredentialAuth(request *types.APIContext, credential string, prevCluster *v32.Cluster) error {
+	if prevCluster != nil && prevCluster.Spec.EKSConfig.AmazonCredentialSecret == credential {
```

Comment:
It doesn't look right that we're checking `EKSConfig.AmazonCredentialSecret` for GKE/Azure cases as well (this is called from `validateAKSConfig` and `validateGKEConfig`.

Reply:
Dang it. What a terrible error. I will fix this.

Reply:
Updated.